app/code/Magento/Sales/Model/Order.php

```diff
@@ -1692,6 +1692,22 @@ public function getAllStatusHistory()
         return $history;
     }
 
+    /**
+     * Return collection of not visible on frontend order status history items.
+     *
+     * @return array
+     */
+    public function getAdminVisibleStatusHistory(): array
```

Comment:
Please move this logic to some ViewModel and use it in the view file.

Reply:
> Please move this logic to some ViewModel and use it in the view file.

@ihor-sviziev it's a good idea! I'll change my code. Thanks!